src/Components/Sidebar/Workspace.tsx

```diff
@@ -1,71 +1,114 @@
-import React from 'react'
+// eslint-disable
+
+import React, { useEffect, useState } from 'react'
 import TreeView from '@material-ui/lab/TreeView'
 import TreeItem from '@material-ui/lab/TreeItem'
 import ExpandMoreIcon from '@material-ui/icons/ExpandMore'
 import ChevronRightIcon from '@material-ui/icons/ChevronRight'
 import { FcGraduationCap, FcAssistant } from 'react-icons/fc'
 import './Workspace.scss'
+import { useQuery, gql } from '@apollo/client'
+import { Button } from '@material-ui/core'
+import { Link } from 'react-router-dom'
+import { makeStyles } from '@material-ui/core/styles'
+import WorkspaceCustom from './WorkspaceCustom'
+
+const useStyles = makeStyles((theme) => ({
+  link: {
+    color: 'black',
+    textDecoration: 'none',
+  },
+}))
+
+const GET_WORKSPACES = gql`
+  query allWorkspaces($input: InputWorkspaceGet!) {
+    allWorkspaces(input: $input) {
+      id
+      title
+      isSchoolWorkspace
+      feed {
+        id
+        feedName
+        messages {
+          id
+          content
+          userId
+          likes
+          dislikes
+          comments {
+            id
+            content
+            userId
+            createdAt
+          }
+        }
+      }
+      assets {
+        id
+        assetName
+      }
+      visio
+    }
+  }
+`
 
 const Workspace: React.FC = () => {
+  const { loading, error, data } = useQuery(GET_WORKSPACES, {
+    variables: {
+      input: {
+        isSchoolWorkspace: true,
+        schoolId: '2',
+      },
+    },
+  })
+  const [workspace, setWorkspace] = useState([])
+  const classes = useStyles()
+  useEffect(() => {
+    if (data) {
+      setWorkspace(data.allWorkspaces)
+    }
+  }, [data])
+
+  if (loading) return <p>Loading...</p>
```

Comment:
it would be nice to add a loader with a react library

I did this on my side: 

const FallBackContainer = styled.div`
  position: fixed;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
`

  if (loading) {
    return (
      <FallBackContainer>
        <Loader type="BallTriangle" color="#000000" height={80} width={80} />
      </FallBackContainer>
    )
  }

to make everything identical on the project at the loader level

Reply:
I used the react-loader-spinner library with the "Ball-Triangle" loader